csodiaq/identifier/scoringFunctions.py

```diff
@@ -0,0 +1,91 @@
+import numpy as np
+from scipy.spatial.distance import cosine
+import matplotlib.pyplot as pyplot
+from csodiaq.identifier.matchingFunctions import eliminate_low_count_matches
+
+def score_library_to_query_matches(matches):
+    scoreDf = matches\
+                .groupby(["libraryIdx", "queryIdx"])\
+                .apply(lambda x: (
+                    calculate_cosine_similarity_score(x["libraryIntensity"], x["queryIntensity"]),
+                    calculate_macc_score(x["libraryIntensity"], x["queryIntensity"])
+                ))\
+                .reset_index(name="scores")
+    scoreDf["cosineScore"],scoreDf["maccScore"] = zip(*scoreDf["scores"])
+    del scoreDf["scores"]
+    return scoreDf.sort_values("maccScore", ascending=False).reset_index(drop=True)
+
+def calculate_cosine_similarity_score(vectorA, vectorB):
+    return 1 - cosine(vectorA, vectorB)
```

Comment:
I played around with doing this, but the formatting and outputs were not as straightforward as desired. It produces a predictive matrix rather than a score. It's possible I could tweak this further to make it work in such a way, but I don't think it would look as clean.